### ISSUE TITLE
box is printed in the top of terminal

### ISSUE THREAD

**Issue by longbeardmen** [piotrmurach/tty-box#1]
I use default demo code to test

```
require 'tty-box'

box = TTY::Box.frame(
  width: 30,
  height: 10,
  align: :center,
  padding: 3
) do
  "Drawing a box in terminal emulator"
end

print box
```

I'v been trying to run that in `irb`or by `ruby file.rb` command execution.
Result is always the same: code ignores current line in terminal, override previous history and prints box in the top of the console window

environment:
ruby version is 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]
iTem version is 3.1.7

**Comment by piotrmurach** [comment]
Hey Roman!

Thanks for trying out this gem!

This is the expected behaviour by design. Why is that? This is so that you can absolute position boxes inside your terminal window, overlay one over the other, create a dashboard of many panels aligned next to each other. I've extracted this library from a terminal game I'm building where I needed to create a frame around various elements of the game. Creating this kind of things based on the relative place of the cursor would be a hard proposition. So you can say that I've satisfied my needs first.

If you're looking into creating a frame around your data in irb then at the moment this isn't possible without absolutely positioning the box. I would be happy to explore possibility to handle the scenario whereby if the `:top`, `:left` locations are provided, the box is drawn with absolute coordinates, otherwise the box is drawn around the content without any positioning.

Do you have time to submit PR?

**Comment by longbeardmen** [comment]
Hm, ok, agreed. for `ruby file.rb` it's ok, but this in irb it looks so ugly. Is it expected too?
https://www.dropbox.com/s/eg4xojj5r41976t/Screen%20Shot%202018-07-27%20at%2011.57.11%20AM.png?dl=0

**Comment by piotrmurach** [comment]
What do you mean the box looks ugly? It looks the same regardless whether you print it the console or script. The print call does exactly the same thing regardless of where you call it. I think this is irrelevant to this discussion. Every single character from the frame is positioned inside a terminal window. This is the expected behaviour. 

However, I'm happy to accomodate your use case, but I'm not 100% sure what do you want to achieve? As I suggested previously, we could look into printing the box around content without positioning it anywhere in the window. However, this will mean you cannot for example, center a box in the middle of your screen. You will need to add however many spaces yourself to nudge to box to the center. Your call what you want to do, just let me know?

**Comment by longbeardmen** [comment]
so current behavior make me to clear console manually and this's not expected:
if I don't clear console it just overlap box upon previous `irb` history, but normal behavior is when your console history shifts to make enough place for new pasted stuff

**Comment by piotrmurach** [comment]
Your initial description made it super clear what the issue is. Repeating examples with issues in irb doesn't progress anything. Therefore I would encourage yourself to focus on the solution to the issue rather than 'educating' me about how it doesn't work in irb session. Otherwise, we will be stuck in an eternal loop 😄 

So let me restate what I've said previously:
> If you're looking into creating a frame around your data in irb then at the moment this isn't possible without absolutely positioning the box. I would be happy to explore possibility to handle the scenario whereby if the :top, :left locations are provided, the box is drawn with absolute coordinates, otherwise the box is drawn around the content without any positioning.
> Do you have time to submit PR?

**Comment by piotrmurach** [comment]
@longbeardmen Don't be like that! I'm not ignoring the issue and you have 100% valid point. However, I would like to hear what is your opinion on the solution? It would be even better if you have time to help out and implement a suggestion and submit PR?

**Comment by piotrmurach** [comment]
@longbeardmen The truth be told, I wanted to work with yourself to get this behaviour implemented, I think this is a good opportunity for collaboration. I'm rather busy but equally, I'm always happy and try my best to encourage other people to jump in and contribute their ideas or code. So don't feel discouraged by my responses because they are all meant to get to help to resolve the issue and provide solution. Anyhow, thanks for rasing this issue!

**Comment by piotrmurach** [comment]
@longbeardmen Released `v0.2.0` that prints the box in irb session as you would expect provided no positioning coordinates are supplied. Let me know if this works well for you!
